File: Facility/Well.cpp
/**
 * @file Well.hpp
 * @author 13515125/Muhammad Rafid Amrullah, 13517092/Michael Ray, 13517113/Juro Sutantra, 13517137/Vincent Budianto
 * @brief file berisi definisi class Well
 * @version 0.1
 * @date 2019-03-20
 */

/* NIM/Nama  : 13515125/Muhammad Rafid Amrullah, 13517092/Michael Ray, 13517113/Juro Sutantra, 13517137/Vincent Budianto
 * Nama file : Well.hpp
 * Topik     : Tugas Besar 1 IF2210 - Pemograman Berorientasi Objek
 * Tanggal   : 20 Maret 2019
 * Deskripsi : Header Well.cpp */

#include <string.h>

#include "Well.hpp"

/**
 * @brief Construct a new Well object
 * 
 */
Well::Well(){
	
}

/**
 * @brief Destroy the Well object
 * 
 */
Well::~Well(){
	
}

/**
 * @brief Method to render the Facility type character to map
 * 
 * @return char 
 */
char Well::render(){
	return 'W';
}

File: Java/model/facility/Well.java
/**
 * Well.java
 * <p>
 * File berisi interface Well
 * </p>
 * @author 13515125/Muhammad Rafid Amrullah, 13517113/Juro Sutantra, 13517137/Vincent Budianto
 * @version 2.0
 * @since 2019-04-21
 */

package facility;

import java.io.*;
import java.lang.*;
import java.util.*;

/**
 * <p>
 * Class for rendering object to Well
 * </p>
 */
public class Well extends Facility
{
	/**
	 * Well constructor
	 * @param y
	 * @param x
	 */
	public Well(int y, int x){
		setY(y);
		setX(x);
	}

	/**
	 * Render Well to map
	 * @return Character
	 */
	public Character render(){
		return 'W';
	}
}
